bricks/r13n/hooks/pre_gen.dart

```diff
@@ -1,17 +1,31 @@
 import 'dart:async';
 import 'dart:convert';
+import 'dart:io' as io;
 import 'dart:io';
 
 import 'package:mason/mason.dart';
-
 import 'package:path/path.dart' as path;
+import 'package:pubspec_parse/pubspec_parse.dart';
 import 'package:yaml/yaml.dart';
 
+typedef Exit = void Function(int);
+
 Future<void> run(HookContext context) async => preGen(context);
 
-Future<void> preGen(HookContext context) async {
-  final configuration = await R13nConfiguration.read();
+Future<void> preGen(
+  HookContext context, {
+  void Function(io.Directory) ensureRuntimeCompatibility =
```

Comment:
Define a typedef for this? So it can be reused in the tests as well? Makes it easier if we ever have a change on the definition of it

Reply:
The typedef would only be used in one place though as far as I can tell. In the tests it's inlined atm:

```dart
preGen(..., ensureRuntimeCompatibility: (_) {})
```